### PR TITLE
add pivot news story

### LINKED ISSUE #576: add "news" item for "tiktok pivot"

### PR DESCRIPTION
**Pull request by akhrail1996**
closes #576

### CODE REVIEW

File: lib/t/news.ex [hunk 1]
```diff
@@ -14,12 +14,177 @@ defmodule T.News do
...
+              "corner_radius" => 0,
+              "alignment" => 0
+            }
+          ],
+          "size" => [375, 667]
+        },
+        %{
+          "blurred" => %{
+            "proxy" =>
+              "https://d3r9yicn85nax9.cloudfront.net/Phqo20PDBNkUCjsdts6lDMc5JD_TxhpXtlCaB6bp3LM/fit/1200/0/sm/0/aHR0cHM6Ly9zaW5jZS13aGVuLWFyZS15b3UtaGFwcHkuczMuYW1hem9uYXdzLmNvbS81Y2ZiZTk2Yy1lNDU2LTQzYmItOGQzYS05OGU4NDljMDBkNWQ",
```
Comment (ruslandoga) [inline lib/t/news.ex:172]: should be dynamic

File: lib/t/news.ex [hunk 2]
```diff
@@ -14,12 +14,177 @@ defmodule T.News do
...
+              "position" => [24.0, 248.0],
+              "corner_radius" => 0,
+              "alignment" => 0
+            },
+            %{
+              "zoom" => 1.0,
+              "value" =>
+                dgettext(
+                  "news",
+                  "К новостям 👉"
```
Comment (ruslandoga) [inline lib/t/news.ex:36]: a bit abrupt

File: lib/t/news.ex [hunk 3]
```diff
@@ -14,12 +14,177 @@ defmodule T.News do
...
+              "zoom" => 1.0,
+              "value" => dgettext("news", "Привет! 👋"),
+              "position" => [24.0, 180.0],
+              "corner_radius" => 0,
+              "alignment" => 0
+            },
+            %{
+              "zoom" => 1.0,
+              "value" =>
+                dgettext("news", "Это новый Since -\nпространство для\nинтересных людей."),
```
Comment (ruslandoga) [inline lib/t/news.ex:26]: did old Since have a different mission? It's not clear that it did.

File: lib/t/news.ex [hunk 4]
```diff
@@ -14,12 +14,177 @@ defmodule T.News do
...
+            %{
+              "zoom" => 1.0,
+              "value" => dgettext("news", "Стикеры-контакты 🔥"),
+              "position" => [24.0, 180.0],
+              "corner_radius" => 0,
+              "alignment" => 0
+            },
+            %{
+              "zoom" => 1.0,
+              "value" => dgettext("news", "Добавляй контакты\nв свою Историю."),
```
Comment (ruslandoga) [inline lib/t/news.ex:57]: why capitalized Story?

File: lib/t/news.ex [hunk 5]
```diff
@@ -14,12 +14,177 @@ defmodule T.News do
...
           "labels" => [
             %{
-              "size" => [247.3, 44],
-              "value" => dgettext("news", "Мы стали совсем другими"),
-              "center" => [142.33, 317.83]
+              "zoom" => 1.0,
+              "value" => dgettext("news", "Привет! 👋"),
+              "position" => [24.0, 180.0],
+              "corner_radius" => 0,
+              "alignment" => 0
```
Comment (ruslandoga) [inline lib/t/news.ex:21]: `alignment`, `corner_radius`, `zoom`, when `0`, should be omitted.

File: priv/gettext/en/LC_MESSAGES/news.po [hunk 6]
```diff
@@ -11,7 +11,72 @@ msgstr ""
...
 #: lib/t/news.ex:18
-msgid "Мы стали совсем другими"
-msgstr ""
+msgid "Привет! 👋"
+msgstr "Hi! 👋"
+
+#, elixir-autogen, elixir-format
+#: lib/t/news.ex:136
+msgid "Создай приватную страницу,\nона будет видна только\nтвоим мэтчам."
+msgstr "Create a private page,\nit will be visible\nonly to your matches."
```
Comment (ruslandoga) [inline priv/gettext/en/LC_MESSAGES/news.po:57]: do we actually have any english speakers btw? Either way, I'd say something like "only your matches would see".

File: priv/gettext/en/LC_MESSAGES/news.po [hunk 7]
```diff
@@ -11,7 +11,72 @@ msgstr ""
...
+#: lib/t/news.ex:112
+msgid "Это ещё не всё 👉"
+msgstr "That's not all 👉"
+
+#, elixir-autogen, elixir-format
+#: lib/t/news.ex:26
+msgid "Это новый Since -\nпространство для\nинтересных людей."
+msgstr "This is the new Since app -\na space for\ninteresing people."
+
+#, elixir-autogen, elixir-format, fuzzy
```
Comment (ruslandoga) [inline priv/gettext/en/LC_MESSAGES/news.po:79]: fuzzy

File: test/t_web/channels/feed_channel_test.exs [hunk 8]
```diff
@@ -444,24 +444,8 @@ defmodule TWeb.FeedChannelTest do
...
-                         "center" => [142.33, 317.83]
-                       }
-                     ],
-                     "size" => [428, 926]
-                   }
-                 ]
-               }
-             ]
+      assert [_first_news_item = %{id: 1, story: story}] = news
+      assert length(story) == 6
```
Comment (ruslandoga) [inline test/t_web/channels/feed_channel_test.exs:448]: this is not testing much, you need to at least keep the response structure test

**Review by ruslandoga — CHANGES_REQUESTED** [review]

File: lib/t/news.ex (hunk 1 above)
Comment (akhrail1996) [reply lib/t/news.ex:172]: should it use sth like this: https://github.com/getsince/test3/blob/master/lib/t_web/views/view_helpers.ex#L60?

File: lib/t/news.ex (hunk 2 above)
Comment (akhrail1996) [reply lib/t/news.ex:36]: Abrupt meaning sudden or rude?
image

File: lib/t/news.ex (hunk 3 above)
Comment (akhrail1996) [reply lib/t/news.ex:26]: yeah, the key difference in this pivot in audio -> contacts, should articulate it better

File: lib/t/news.ex (hunk 4 above)
Comment (akhrail1996) [reply lib/t/news.ex:57]: kinda putting greater emphasis on the Story, we did in some other places as well

File: lib/t/news.ex (hunk 1 above)
Comment (ruslandoga) [reply lib/t/news.ex]: yes

File: lib/t/news.ex (hunk 2 above)
Comment (ruslandoga) [reply lib/t/news.ex]: sudden

File: lib/t/news.ex [hunk 9]
```diff
@@ -10,16 +10,137 @@ defmodule T.News do
...
+        %{
+          "background" => %{"color" => "#111010"},
+          "labels" => [
+            %{
+              "value" => dgettext("news", "Привет! 👋"),
+              "position" => [24.0, 180.0],
+              "background_fill" => "#F97EB9"
+            },
+            %{
+              "value" => dgettext("news", "Это Since -\nпространство для\nинтересных людей."),
```
Comment (ruslandoga) [inline lib/t/news.ex:22]: что поменялось?

File: lib/t/news.ex [hunk 10]
```diff
@@ -10,16 +10,137 @@ defmodule T.News do
...
+              "position" => [24.0, 248.0],
+              "background_fill" => "#F97EB9"
+            },
+            %{
+              "value" => dgettext("news", "Обновления 👉"),
```
Comment (ruslandoga) [inline lib/t/news.ex:27]: выше нужно написать, что поменялось, тут нужно указать, что будет дальше. Можно использовать больше одного слова.

File: lib/t/news.ex (hunk 4 above)
Comment (ruslandoga) [reply lib/t/news.ex:42]: it kinda sucks

File: lib/t/news.ex [hunk 11]
```diff
@@ -10,16 +10,137 @@ defmodule T.News do
...
-              "center" => [142.33, 317.83]
+              "value" => dgettext("news", "Стикеры-контакты 🔥"),
+              "position" => [24.0, 180.0]
+            },
+            %{
+              "value" => dgettext("news", "Добавляй контакты\nв свою Историю."),
+              "position" => [24.0, 248.0]
+            },
+            %{
+              "value" => dgettext("news", "Нажми 👇"),
```
Comment (ruslandoga) [inline lib/t/news.ex:46]: нажми, и? нужно указать что это, "Вот наши"

File: lib/t/news.ex [hunk 12]
```diff
@@ -10,16 +10,137 @@ defmodule T.News do
...
+                  "news",
+                  "На нее можно поместить\nсвой контакт и что-то\nболее личное про тебя."
+                ),
+              "position" => [24.0, 322.0]
+            },
+            %{
+              "value" =>
+                dgettext(
+                  "news",
+                  "Выглядит вот так 👉"
```
Comment (ruslandoga) [inline lib/t/news.ex:122]: как? если нужно свайпать, об этом нужно написать

**Review by ruslandoga — CHANGES_REQUESTED** [review]
Sorry, I can't approve it. You can merge it without my approval.

File: lib/t_web/channels/feed_channel.ex [hunk 13]
```diff
@@ -54,7 +54,7 @@ defmodule TWeb.FeedChannel do
       |> Matches.list_archived_matches()
       |> render_matches(screen_width)
 
-    news = News.list_news(user_id)
+    news = News.list_news(user_id) |> render_news(screen_width)
```
Comment (ruslandoga) [inline lib/t_web/channels/feed_channel.ex:57]: no

File: lib/t_web/channels/feed_channel.ex [hunk 14]
```diff
@@ -541,4 +541,8 @@ defmodule TWeb.FeedChannel do
   defp render_interaction(interaction) do
     render(MatchView, "interaction.json", interaction: interaction)
   end
+
+  defp render_news(news, screen_width) do
+    Enum.map(news, fn n -> render(FeedView, "news.json", news: n, screen_width: screen_width) end)
```
Comment (ruslandoga) [inline lib/t_web/channels/feed_channel.ex:546]: no

File: lib/t_web/views/feed_view.ex [hunk 15]
```diff
@@ -21,6 +21,13 @@ defmodule TWeb.FeedView do
     %{"profile" => profile, "call" => CallView.render("call.json", call: call)}
   end
 
+  def render("news.json", %{news: news, screen_width: screen_width}) do
+    news
+    |> Map.update!(:story, fn story ->
+      ViewHelpers.postprocess_story(story, screen_width, _env = :feed)
```
Comment (ruslandoga) [inline lib/t_web/views/feed_view.ex:27]: no

**Pull request merged**